contracts/SportMarkets/Parlay/ParlayLP/ParlayAMMLiquidityPool.sol

```diff
@@ -0,0 +1,758 @@
+
+        // new user enters the pool
+        if (balancesPerRound[round][msg.sender] == 0 && balancesPerRound[nextRound][msg.sender] == 0) {
+            require(usersCurrentlyInPool < maxAllowedUsers, "Max amount of users reached");
+            usersPerRound[nextRound].push(msg.sender);
+            usersCurrentlyInPool = usersCurrentlyInPool + 1;
+        }
+
+        balancesPerRound[nextRound][msg.sender] += amount;
+
+        allocationPerRound[nextRound] += amount;
+        totalDeposited += amount;
+
+        if (address(stakingThales) != address(0)) {
+            stakingThales.updateVolume(msg.sender, amount);
+        }
+
+        emit Deposited(msg.sender, amount, round);
+    }
+
+    /// @notice get sUSD to mint for buy and store market as trading in the round
+    /// @param market to trade
+    /// @param amountToMint amount to get for mint
+    function commitTrade(address market, uint amountToMint)
+        external
+        nonReentrant
+        whenNotPaused
+        onlyAMM
+        roundClosingNotPrepared
+    {
+        require(started, "Pool has not started");
+        require(amountToMint > 0, "Can't commit a zero trade");
+
+        amountToMint = _transformCollateral(amountToMint);
+        // add 1e-6 due to rounding issue, will be sent back to AMM at the end
+        amountToMint = needsTransformingCollateral ? amountToMint + 1 : amountToMint;
+
+        uint marketRound = getMarketRound(market);
+        parlayMarketRound[market] = marketRound;
+        address liquidityPoolRound = _getOrCreateRoundPool(marketRound);
+        if (marketRound == round) {
+            sUSD.safeTransferFrom(liquidityPoolRound, address(parlayAMM), amountToMint);
+        } else if (marketRound > round) {
+            uint poolBalance = sUSD.balanceOf(liquidityPoolRound);
+            if (poolBalance >= amountToMint) {
+                sUSD.safeTransferFrom(liquidityPoolRound, address(parlayAMM), amountToMint);
+            } else {
+                uint differenceToLPAsDefault = amountToMint - poolBalance;
+                _depositAsDefault(differenceToLPAsDefault, liquidityPoolRound, marketRound);
+                sUSD.safeTransferFrom(liquidityPoolRound, address(parlayAMM), amountToMint);
+            }
+        } else {
+            require(marketRound == 1, "InvalidRound");
+            _provideAsDefault(amountToMint);
+        }
+
+        tradingMarketsPerRound[marketRound].push(market);
+        isTradingMarketInARound[marketRound][market] = true;
+    }
+
+    function transferToPool(address _market, uint _amount) external nonReentrant whenNotPaused roundClosingNotPrepared {
+        uint marketRound = getMarketRound(_market);
+        address liquidityPoolRound = marketRound <= 1 ? defaultLiquidityProvider : _getOrCreateRoundPool(marketRound);
+        sUSD.transferFrom(address(parlayAMM), liquidityPoolRound, _amount);
+        marketAlreadyExercisedInRound[marketRound][_market] = true;
```

Comment:
I am not sure this is place where this needs to be done. This should be set to true if:
1. There are no more markets on the parlay to check (all markets resolved and exercised)
2. User has won the parlay
3. All markets on the parlay are lost